src/avx512-common-qsort.h

```diff
@@ -85,6 +85,13 @@
 #define X86_SIMD_SORT_FINLINE static
 #endif
 
+#if __GNUC__ >= 8
```

Comment:
I noted that you use __GNUC__ to adjust gcc version, but oceanbase downloads its own gcc in the processing of building, not the system gcc. So when oceanbase compiles, it will get the wrong __GNUC__. For example, my system gcc is version 11.2. The __GNUC__ will be 11, but actually, oceanbase uses its own gcc to build itself, maybe gcc 8.5 or others. Maybe we can add "#pragma GCC diagnostic ignored "-Wunknown-pragmas" " in this file. Do you think it is ok?

Reply:
Is it not possible to use `-Wno-unknown-pragmas` as a compiler flag when building this as part of oceanbase?

Reply:
Ok, we can have a try, not sure oceanbase can accept it or not. Thanks

Reply:
@r-devulap we have added the flag into the Oceanbase PR to see if they accept it. Thanks